api/server/router/container/container_routes.go

```diff
@@ -498,8 +499,8 @@ func (s *containerRouter) postContainersCreate(ctx context.Context, w http.Respo
 		}
 	}
 	if hostConfig != nil && versions.LessThan(version, "1.41") {
-		// Older clients expect the default to be "host"
-		if hostConfig.CgroupnsMode.IsEmpty() {
+		// Older clients expect the default to be "host" on cgroup v1 hosts
+		if hostConfig.CgroupnsMode.IsEmpty() && !cgroups.IsCgroup2UnifiedMode() {
 			hostConfig.CgroupnsMode = container.CgroupnsMode("host")
```

Comment:
IIUC, cgroupv2 does not support "non-private"?

So, should we handle this when creating the actual container (as well / instead) ? Thinking of a scenario where a container was created with an older API version and the host is running cgroupv1, then the host is switched to cgroupv2. In that case existing containers would no longer be functional?

Reply:
v2 still supports `--cgroupns=host`, we just switched the default

Reply:
Ignoring the API version means that we change behaviour of an already release API version?

Reply:
Yes and no, depending on the definition of the previous behavior.
Yes if we define the previous behavior as `always defaults to "host"`".  
No if `defaults to the most typical config, depending on the cgroup version`.

As there hadn't been support for cgroup v2 in previous versions, it is hard to say which definition is correct.
In this PR, I adopted the latter definition.

Reply:
Since we haven't supported cgroupv2 before, I think there is no expectation that cgroupv2 default will be different for older API version